client/deployment_delete.go

```diff
@@ -0,0 +1,45 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// DeleteDeploymentResponse defines the response the Vercel API returns when a deployment is deleted.
+type DeleteDeploymentResponse struct {
+	State string `json:"state"`
+	UID   string `json:"uid"`
+}
+
+// DeleteDeployment deletes a deployment within Vercel.
+func (c *Client) DeleteDeployment(ctx context.Context, deploymentURL string, teamID string) (r DeleteDeploymentResponse, err error) {
+	deploymentID := "ignored" //This is ignored if URL query parameter is passed as per https://vercel.com/docs/rest-api#endpoints/deployments/delete-a-deployment
+	url := fmt.Sprintf("%s/v13/deployments/%s", c.baseURL, deploymentID)
+	req, err := http.NewRequest(
+		"DELETE",
+		url,
+		nil,
+	)
+	if err != nil {
+		return r, err
+	}
+
+	// Add query parameters
+	q := req.URL.Query()
+	q.Add("teamId", teamID)
```

Comment:
I think teamID should be conditional, so this should only be added if there is a teamID present? Or does `req.URL.Query()` handle this?

Reply:
you're right, changing this